packages/app/src/components/__tests__/DBRowTable.test.tsx

```diff
@@ -0,0 +1,35 @@
+import { appendSelectWithPrimaryAndPartitionKey } from '@/components/DBRowTable';
+
+describe('appendSelectWithPrimaryAndPartitionKey', () => {
```

Comment:
I think there's a few more edge cases we might want to account for that I just realized:

```ts
  it('should extract map columns', () => {
    const result = appendSelectWithPrimaryAndPartitionKey('col1, col2', `map['key']`, `map2['key'], map1['key3']`);
    expect(result).toEqual({
      additionalKeysLength: 3,
      select: `col1,col2,map['key'],map2['key'],map1['key3']`,
    });
  });

  it('should extract json columns', () => {
    const result = appendSelectWithPrimaryAndPartitionKey('col1, col2', `json.b`, `json.a`);
    expect(result).toEqual({
      additionalKeysLength: 2,
      select: `col1,col2,json.a,json.b`,
    });
  });
```

I expect this to be a bit more rare to be honest, I think it's okay if the function bails instead as well.